adal/oauth2_client.py

```diff
@@ -138,7 +138,7 @@ def _parse_id_token(self, encoded_token):
                 self._log.warn('The returned id_token could not be base64 url safe decoded.')
                 return
 
-            id_token = json.loads(b64_decoded.decode())
+            id_token = json.loads(b64_decoded.decode('utf-8'))

```

Comment:
I did take that into consideration. Current customers either did not (yet) meet this error so they don't rely on the current behavior, or they had already been forced to come up with code to deal with the `None` return value etc.. In the latter case, their code still runs, just less needed, because this fix decodes properly.